module-system/sov-modules-stf-template/src/lib.rs

```diff
@@ -99,8 +99,7 @@ where
             .expect("jellyfish merkle tree update must succeed");
 
         let mut working_set = WorkingSet::new(self.current_storage.clone());
-
-        self.runtime.end_slot_hook(&mut working_set);
+        self.runtime.end_slot_hook(root_hash, &mut working_set);
```

Comment:
@neysofu we will need to update the `non-JMT` data in the `end_slot_hook` but with the latest state root hash. This raises the question of when we save the `non-JMT` data.

Reply:
Yup, as agreed during our call. I think we have a plan to tackle this.